### PR TITLE
RavenDB-18991 - Fix QueriesShouldFailoverIfIndexIsCompactingCluster  …

### PR DESCRIPTION
**Pull request by shaharhikri**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-18991/SlowTestsIssuesRavenDB18554QueriesShouldFailoverIfIndexIsCompactingCluster

### Additional description

Fixing QueriesShouldFailoverIfIndexIsCompactingCluster test failure.

### Type of change

- Bug fix

### How risky is the change?

- Low 

### Backward compatibility

- Not relevant

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing 

- Tests have been added that prove the fix is effective or that the feature works

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

**Review by karmeli87 — APPROVED** [review]

File: src/Raven.Client/ServerWide/Operations/CompactDatabaseOperation.cs [hunk 1]
```diff
@@ -11,23 +11,29 @@ namespace Raven.Client.ServerWide.Operations
...
     {
+        private readonly string _selectedNode;
         private readonly CompactSettings _compactSettings;
 
         public CompactDatabaseOperation(CompactSettings compactSettings)
         {
             _compactSettings = compactSettings ?? throw new ArgumentNullException(nameof(compactSettings));
         }
 
+        internal CompactDatabaseOperation(CompactSettings compactSettings, string selectedNode) : this(compactSettings)
```
Comment (ppekrol) [inline src/Raven.Client/ServerWide/Operations/CompactDatabaseOperation.cs]: we call it `nodeTag` in other commands

File: src/Raven.Client/ServerWide/Operations/CompactDatabaseOperation.cs [hunk 2]
```diff
@@ -11,23 +11,29 @@ namespace Raven.Client.ServerWide.Operations
 {
     public class CompactDatabaseOperation : IServerOperation<OperationIdResult>
     {
+        private readonly string _selectedNode;
```
Comment (ppekrol) [inline src/Raven.Client/ServerWide/Operations/CompactDatabaseOperation.cs]: we call it `nodeTag` in other commands

File: src/Raven.Client/ServerWide/Operations/CompactDatabaseOperation.cs [hunk 3]
```diff
@@ -11,23 +11,29 @@ namespace Raven.Client.ServerWide.Operations
...
-            return new CompactDatabaseCommand(conventions, context, _compactSettings);
+            return new CompactDatabaseCommand(conventions, context, _compactSettings, _selectedNode);
         }
 
         private class CompactDatabaseCommand : RavenCommand<OperationIdResult>
         {
             private readonly BlittableJsonReaderObject _compactSettings;
 
-            public CompactDatabaseCommand(DocumentConventions conventions, JsonOperationContext context, CompactSettings compactSettings)
+            public CompactDatabaseCommand(DocumentConventions conventions, JsonOperationContext context, CompactSettings compactSettings, string selectedNode)
```
Comment (ppekrol) [inline src/Raven.Client/ServerWide/Operations/CompactDatabaseOperation.cs:36]: we call it `nodeTag` in other commands

File: src/Raven.Client/ServerWide/Operations/CompactDatabaseOperation.cs (hunk 1 above)
Comment (shaharhikri) [reply src/Raven.Client/ServerWide/Operations/CompactDatabaseOperation.cs]: Done

File: src/Raven.Client/ServerWide/Operations/CompactDatabaseOperation.cs (hunk 2 above)
Comment (shaharhikri) [reply src/Raven.Client/ServerWide/Operations/CompactDatabaseOperation.cs]: Done

File: src/Raven.Client/ServerWide/Operations/CompactDatabaseOperation.cs (hunk 3 above)
Comment (shaharhikri) [reply src/Raven.Client/ServerWide/Operations/CompactDatabaseOperation.cs:36]: Done

**Pull request merged**